### PR TITLE
feat(dropdown): added secondary split toggle

### LINKED ISSUE #4798: Split Action Button - Add secondary button styling
Can we add in a variation of the split action button that uses our secondary button styling?  We support primary styling for the split action, but not secondary. Support for secondary styling would be great to differentiate from other primary buttons on the page, while also differentiating/emphasizing the split action button from the basic dropdown/select styling. https://www.patternfly.org/v4/components/dropdown#split-button-primary-action

### PR DESCRIPTION
**Pull request by mattnolting**
Closes #4798

### CODE REVIEW

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @mattnolting !

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 1]
```diff
@@ -218,6 +229,7 @@ $pf-c-dropdown--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "
 
   &.pf-m-expanded {
     --pf-c-dropdown__toggle--m-split-button--m-primary--child--BackgroundColor: var(--pf-c-dropdown__toggle--m-split-button--m-primary--child--m-expanded--BackgroundColor);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss:231]: Can you leave a comment here to update this var name to `--pf-c-dropdown--m-expanded__toggle--m-split-button--m-primary--child--BackgroundColor` like you've done with the secondary var?

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 2]
```diff
@@ -218,6 +229,7 @@ $pf-c-dropdown--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "
...
+    --pf-c-dropdown__toggle--m-secondary--m-split-button--child--before--BorderWidth: var(--pf-c-dropdown--m-expanded__toggle--m-secondary--m-split-button--button--before--BorderWidth);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss:232]: should this be `--child`?

```suggestion
    --pf-c-dropdown__toggle--m-secondary--m-split-button--child--before--BorderWidth: var(--pf-c-dropdown--m-expanded__toggle--m-secondary--m-split-button--child--before--BorderWidth);
```

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 3]
```diff
@@ -507,6 +519,52 @@ $pf-c-dropdown--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "
...
+    }
+
+    > * {
+      &::before {
+        border: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--child--before--BorderWidth) solid var(--pf-c-dropdown__toggle--m-secondary--m-split-button--child--before--BorderColor);
+      }
+
+      &:hover,
+      &:focus,
+      &:active {
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss:538]: I wonder if we should have unique vars for these states - seems like we do with the other variations.

**Review by mcoker — COMMENTED** [review]
Looks great! Just a few small things.

**Comment by mattnolting** [comment]
@mcoker updated!

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 4]
```diff
@@ -507,6 +523,64 @@ $pf-c-dropdown--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "
...
 
+  &.pf-m-secondary.pf-m-split-button:not(.pf-m-disabled) {
+    --pf-c-dropdown__toggle-button--Color: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--Color);
+
+    &:hover {
+      --pf-c-dropdown__toggle--m-secondary--hover--before--BorderWidth: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--hover--before--BorderWidth);
+    }
+
+    &:focus {
+      --pf-c-dropdown__toggle--m-secondary--hover--before--BorderWidth: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--focus--before--BorderWidth);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss:534]:
```suggestion
      --pf-c-dropdown__toggle--m-secondary--focus--before--BorderWidth: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--focus--before--BorderWidth);
```

File: src/patternfly/components/Dropdown/dropdown.scss [hunk 5]
```diff
@@ -507,6 +523,64 @@ $pf-c-dropdown--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "
...
+    }
+
+    &:active {
+      --pf-c-dropdown__toggle--m-secondary--hover--before--BorderWidth: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--active--before--BorderWidth);
```
Comment (mcoker) [inline src/patternfly/components/Dropdown/dropdown.scss:538]:
```suggestion
      --pf-c-dropdown__toggle--m-secondary--active--before--BorderWidth: var(--pf-c-dropdown__toggle--m-secondary--m-split-button--active--before--BorderWidth);
```

**Review by mcoker — CHANGES_REQUESTED** [review]

**Review by mcoker — APPROVED** [review]
👍👍

**Pull request merged**